packages/flutter_tools/lib/src/ios/simulators.dart

```diff
@@ -325,7 +332,7 @@ class IOSSimulator extends Device {
     bool usesTerminalUi = true,
     bool ipv6 = false,
   }) async {
-    if (!prebuiltApplication) {
+    if (!prebuiltApplication && package is BuildableIOSApp) {
```

Comment:
If it's not prebuilt and not buildable, then how did we get here? Should the 'is' check be an assert?

Reply:
we need the `is` check to make this not an implicit cast, even with the covariant on _setupUpdatedApplicationBundle